src/sentry/api/endpoints/event_owners.py

```diff
@@ -35,11 +35,26 @@ def get(self, request, project, event_id):
         if owners == ProjectOwnership.Everyone:
             owners = []
 
+        serialized_owners = serialize(Actor.resolve_many(owners), request.user, ActorSerializer())
+        # We do so many dict/set casts on these owners that the order is not preserved at all.
+        # Re-order the results according to how the rules are ordered.
+        owner_map = {o["name"]: [o, False] for o in serialized_owners}
+        ordered_owners = []
+        for rule in rules:
+            for o in rule.owners:
+                found = owner_map.get(o.identifier)
+                if found:
+                    ordered_owners.append(found[0])
+                    found[1] = True
+
+        # A litle defensive, just in case there are owners that aren't part of the rules, add them to the end.
```

Comment:
Where would those owners come from? You could also raise/log a message if the ordered_owners length != serialized_owners.

Reply:
Right now I don't think there is any place for this to happen, which is why it's defensive. I can remove it completely and everything will still work fine. I just thought it might avoid headaches later down the line if the underlying functions change.

Reply:
If you use `logging.error()` here it will trigger a sentry error and we'll be able to track down this situtation happening fast. If it doesn't fire in a few days you're probably safe to remove this.

Reply:
OK I'll do that. Thanks.